app/src/main/java/com/palone/planahead/data/database/TaskWithAlerts.kt

```diff
@@ -0,0 +1,12 @@
+package com.palone.planahead.data.database
+
+import androidx.room.Embedded
+import androidx.room.Relation
+import com.palone.planahead.data.database.alert.Alert
+import com.palone.planahead.data.database.task.Task
+
+data class TaskWithAlerts(
+    @Embedded val task: Task,
+    @Relation(parentColumn = "taskId", entityColumn = "taskId")
+    val alerts: List<Alert>
```

Comment:
You should use the most generic type possible - `Iterable<Alert>` would be better. That way we could for example introduce a special version of `getWithAlerts` that returns alerts in batches (in this case very unlikely, but it's a good thing to keep your options open).

Reply:
Iterable can't be used because then the whole TaskWithAlerts class should be marked as Entity, which would break the database ([dosc](https://developer.android.com/training/data-storage/room/relationships#one-to-many))